### ISSUE TITLE
Card: Missing keyboard navigation on clickable Cards

### ISSUE THREAD

**Issue by khmakoto** [OfficeDev/office-ui-fabric-react#11406]
The `Card` component currently has an `onClick` property that makes it possible to interact with it via mouse and touch controls. However, the `Card` itself is not reachable via keyboard navigation, which is an omission since it breaks accessibility within the component. When the `onClick` property is provided, the `Card` should be reachable via keyboard.
